pandas/core/generic.py

```diff
@@ -192,8 +192,9 @@ def __unicode__(self):
 
     def _dir_additions(self):
         """ add the string-like attributes from the info_axis """
-        return set([c for c in self._info_axis
-                    if isinstance(c, string_types) and isidentifier(c)])
+        additions = set([c for c in self._info_axis
+                         if isinstance(c, string_types) and isidentifier(c)])
+        return PandasObject._dir_additions(self).union(additions)
```

Comment:
Not really.  I developed the habit working in `statsmodels` where inheritance chains can get pretty messy.  Happy to change it.